### PR TITLE
Create component command

### PR DESCRIPTION
**Pull request by thiagodemellobueno**
# Command
```kalastatic create-component <objectToCreate> [otherObjects...]```

## Alias
``cc``

## Options
- ```-d, --directory=path/to/where/you/want/```
defaults to ```src/components```

## Usage
```kalastatic cc atoms/links``` 
or ```kalastatic cc atoms/link```
or ```kalastatic cc --directory=path/you/want/the/files/to/go atoms/link atoms/button molecules/article-teaser```

or FTW

```kalastatic cc --directory=path/you/want/the/files/to/go `cat components.txt` ```
assuming the contents of components.txt is:
```
atoms/links
atoms/buttons
molecules/cta
molecules/tout
```

Also simplified basic test… if we want to test conventions-based js concat we should put it in a separate text (behavior changed from expected and was breaking things)…..

Again I’m not super into creating invalid js files just to theoretically simplify an already simple process, I would much rather tell kalastatic.yml which js files to contact in an array.

Creating it as kalastatic create component became unruly, and required a lot of logic in an otherwise simple command, i recommend we make dedicated commands ... si I named this one create-component so if we do create a 'global create' we can just call this one from there.

### CODE REVIEW

**Comment by thiagodemellobueno** [comment]
@soniktrooth @RobLoach

**Comment by thiagodemellobueno** [comment]
Looks to address #364

**Comment by thiagodemellobueno** [comment]
@RobLoach one thing I would want to consider is defaults for --directory

right now its just 'where you called kalastatic' from which I think makes the most sense, but it also kinda means you almost always want to include --directory, would it make sense to assume src/components in the default case or is that just 'our way'

**Comment by thiagodemellobueno** [comment]
Also look PR as documentation!

**Comment by derekderaps** [comment]
Niiiiice!!

File: bin/kalastatic [hunk 1]
```diff
@@ -92,6 +94,51 @@ program
...
+function createComponent(dir,objectToCreate) {
+  var component = objectToCreate.split("/")
+  var basePath = path.join(dir, component[0], component[1])
+  mkdirp(basePath, function(){
+    var scssFile = path.join(basePath, "_"+component[1]+'.scss')
+    var jsonFile = path.join(basePath, component[1]+'.json')
+    var twigFile = path.join(basePath, component[1]+'.twig')
+    createFile(scssFile)
+    createFile(jsonFile)
+    createFile(twigFile)
```
Comment (RobLoach) [inline bin/kalastatic]: Looks like createFile() uses async methods. We'll either want to switch those over to the [writeFileSync](https://nodejs.org/docs/v0.3.1/api/fs.html#fs.writeFileSync) alternative, or handle the asynchronous callbacks.

Would also be neat to give it a base template. For example, give the scss file something like....

``` scss
/**
 * Template Styles
 *
 * This is where the styles go...
 *
 * Styleguide: MyStuff.STUFFS
 */

// Nothing
```
or something.

File: bin/kalastatic [hunk 2]
```diff
@@ -92,6 +94,51 @@ program
...
+  })
+
+}
+program
+  .command('create-component <objectToCreate> [otherObjects...]')
+  .alias('cc')
+  .description('Automates the creation of… things')
+  .option('-d, --directory <path>', 'where to run')
```
Comment (RobLoach) [inline bin/kalastatic]: Nice one :+1:

File: test/fixtures/basic/src/scripts.js.uglify-js.concat [hunk 3]
```diff
@@ -1,11 +0,0 @@
...
-files:
-- script.js
-# Put additional JavaScript in "files".
-mangle: true
----
-
-// This is scripts.js.uglify-js.concat
-var name = 'World'
-var message = 'Hello ,' + name + '!'
-console.log(message)
```
Comment (RobLoach) [inline test/fixtures/basic/src/scripts.js.uglify-js.concat]: I think the basic test should be fixed now. You're right that it was broken before. Bringing in the latest from `master` should fix it.

File: test/index.js [hunk 4]
```diff
@@ -68,3 +70,20 @@ test('cli', function (done) {
     done()
   })
 })
+
+test('createcomponent', function (done) {
+  var options = {
+    cwd: '.'
+  }
+  rimraf.sync('test/fixtures/createcomponent/build/components')
+  exec('bin/kalastatic cc --directory=test/fixtures/createcomponent/build/components atoms/links', options, function (err, stdout, stderr) {
```
Comment (RobLoach) [inline test/index.js]: Really like this test :+1:

Comment (thiagodemellobueno) [inline test/index.js]: Thanks!

File: bin/kalastatic [hunk 5]
```diff
@@ -92,6 +94,51 @@ program
...
+  .parse(process.argv)
+  .action( function(objectToCreate, otherObjects, options) {
+    console.log("kalastatic creating component »»", objectToCreate)
+    var dir = options.directory || "src/components/"
```
Comment (RobLoach) [inline bin/kalastatic]: That's a good idea.

**Comment by RobLoach** [comment]
> @thiagodemellobueno  one thing I would want to consider is defaults for --directory

Let's tackle that in a follow up PR. Nicely done! Merging this one.

**Pull request merged**